dev-tools/validation/check-authz-route-matrix.py

```
@@ -0,0 +1,246 @@
+        for key, child in value.items():
+            if key == "required_live_checks":
+                checks.extend(child if isinstance(child, list) else [child])
+            else:
+                checks.extend(collect_live_checks(child))
+    elif isinstance(value, list):
+        for child in value:
+            checks.extend(collect_live_checks(child))
+    return checks
+
+
+def validate_matrix_document(path: Path) -> tuple[list[str], set[str]]:
```

Ruff (0.15.21)

[warning] Too many branches (32 > 12)

(PLR0912)